### PR TITLE
Test file operations in dolphin

### PR DESCRIPTION
**Pull request by Vogtinator**
This adds file operations to the dolphin test.

- Needles: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/396
- Verification run: http://10.160.67.86/tests/111

### CODE REVIEW

File: tests/x11/dolphin.pm [hunk 1]
```diff
@@ -1,23 +1,74 @@
...
+    assert_screen 'dolphin_new_folder';
+    send_key 'ret';
+    # Enter the new folder
+    send_key 'ret';
+
+    # Context menu: "Create new" -> "Text file"
+    assert_and_click 'dolphin_stuff_empty', 'right';
+    my $create_new = assert_screen 'dolphin_create_new';
+
+    my $lastarea = $create_new->{area}->[-1];
```
Comment (kravciak) [inline tests/x11/dolphin.pm]: You can use `xy` function from lib/caasp_controller.pm for getting coordinates

Comment (Vogtinator) [reply tests/x11/dolphin.pm]: Why is that in caasp_controller? If it were somewhere more fitting I would use it...

File: tests/x11/dolphin.pm [hunk 2]
```diff
@@ -1,23 +1,74 @@
...
+    my $x        = int($lastarea->{x} + $lastarea->{w} / 2);
+    my $y        = int($lastarea->{y} + $lastarea->{h} / 2);
+    mouse_set($x, $y);
+    mouse_click();
```
Comment (asmorodskyi) [inline tests/x11/dolphin.pm]: awesome :)

File: tests/x11/dolphin.pm (hunk 1 above)
Comment (asmorodskyi) [reply tests/x11/dolphin.pm]: you may suggest PR which move this function to utils.pm :)

Comment (Vogtinator) [reply tests/x11/dolphin.pm]: A new parameter "hide_mouse" to assert_and_click probably makes more sense

**Comment by andreasstieger** [comment]
Leap 15.0 maintenance tests:
https://openqa.opensuse.org/tests/705826#step/dolphin/5
https://openqa.opensuse.org/tests/705825#step/dolphin/5

**Comment by andreasstieger** [comment]
Leap 42.3 and 15.0 maintenance testes remain broken. What's the deal @Vogtinator ?
 
https://openqa.opensuse.org/tests/706368#step/dolphin/14
https://openqa.opensuse.org/tests/706340#step/dolphin/14
https://openqa.opensuse.org/tests/706074#step/dolphin/6
https://openqa.opensuse.org/tests/706365#step/dolphin/9

@czerw this may be something for you to deal with?

**Comment by Vogtinator** [comment]
> Leap 42.3 and 15.0 maintenance testes remain broken. What's the deal @Vogtinator ?

Just missing needles - TW has previews turned on by default, which unfortunately means that none of the (many) needles were reusable for 15.0 and 42.3 appears to have slightly different font rendering.

Creating needles in a live instance is a bit cumbersome (it'll get worse once the interactive mode gets removed as part of the next deployment!) so it takes a full test run per needle...

I'll watch out for new failures.

**Comment by czerw** [comment]
@andreasstieger  well, I noticed openSUSE failing tests, restart by @Vogtinator in audit log and new needle each restart. So it was obvious that work is in progress, which is great. But on other side, it makes review confusing. It would be better to mark it with poo, update needles in private instance and create new PR  for Leap needle update.

**Comment by okurz** [comment]
@czerw I guess the main reason is because the interactive mode was removed from current openQA so a local run is not faster to fix many needles.

**Comment by czerw** [comment]
@Vogtinator what is the status of needles for 42.3? There are still failures https://openqa.opensuse.org/tests/707288#step/dolphin/10. Thank you.

**Comment by Vogtinator** [comment]
@czerw Waiting to be merged: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/398

It also needs https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/5395 to work.

**Pull request merged**